ZumaGame.cpp

```diff
@@ -0,0 +1,82 @@
+// code for Zuma Game
+// https://leetcode.com/problems/zuma-game/
+
+#include <iostream>
+#include <list>
+#include <vector>
+using namespace std;
+
+const int kMaxNeed = 6;
+struct Node{
+    char ch;
+    int count;
+    Node(char c, int i) : ch(c), count(i){}
+};
+void CleanBoard(list<Node> &board){
```

Comment:
As the list size is small (<=20), a vector may have better performance, e.g. it's much faster to make a copy during following recursion calls. But we need to avoid the erase operations on a vector.

By using another vector as a stack, we only need to copy the vector at most once. I tried following code, it uses less time.

void CleanBoard(vector<Node> &board) {
  if no Node in board has 'count >= 3', return // fast path, avoid unnecessary copying

  vector<Node> stack;
  stack.reserve(board.size()); // avoid dynamic resize
  for (node in board) {
    if node.ch is same as stack.end, stack.end.count += node.count
    else push node to stack
    if stack.end.count >= 3, pop
  }
  board.swap(stack)
}